pkg/pillar/containerd/containerd.go

```diff
@@ -146,62 +135,19 @@ func CtrRm(containerPath string, silent bool) error {
-	var (
-		ociOpts       []oci.SpecOpts
-		containerOpts []containerd.NewContainerOpts
-		ociSpec       oci.Spec
-		err           error
-	)
-	if CtrdClient == nil {
-		return fmt.Errorf("ctrCreate: Container client is nil")
-	}
-
-	if ctrdCtx == nil {
-		return fmt.Errorf("ctrCreate: Container context is nil")
-	}
-
-	// containerOpts = append(containerOpts, containerd.WithNewSpec(oci.WithImageConfig(ctrdImage)))
-	containerSnapshot := getSnapshotName(containerID)
-	containerOpts = append(containerOpts,
-		containerd.WithImage(ctrdImage),
-		containerd.WithSnapshotter(defaultSnapshotter),
-		containerd.WithNewSnapshot(containerSnapshot, ctrdImage),
-		containerd.WithRuntime("io.containerd.runc.v2", &options.Options{}),
-		containerd.WithSpec(&ociSpec, ociOpts...))
-	container, err := CtrdClient.NewContainer(
-		ctrdCtx,
-		containerID,
-		containerOpts...,
-	)
-	if err != nil {
-		log.Errorf("Could not build new containerd container: %v. %v", containerID, err.Error())
-		return fmt.Errorf("ctrCreate: Could not build new containerd container: %v. %v", containerID, err.Error())
-	}
-	imageConfigJSON, err := getImageInfoJSON(ctrdCtx, ctrdImage)
-	if err != nil {
-		container.Delete(ctrdCtx)
-		log.Errorf("Could not build json of image: %v. %v", ctrdImage.Name(), err.Error())
-		return fmt.Errorf("ctrCreate: Could not build json of image: %v. %v", ctrdImage.Name(), err.Error())
-	}
-	err = createBundle(container, containerSnapshot, imageConfigJSON)
-	if err != nil {
-		CtrRm(containerID, true)
-		log.Errorf("Could not build rootfs of container: %v. %v", containerID, err.Error())
-		return fmt.Errorf("ctrCreate: Could not build rootfs of container: %v. %v", containerID, err.Error())
-	}
-	return nil
-}
-
-// CtrPrepare prepare an existing container
-func CtrPrepare(containerPath string, ociFilename string) error {
-	log.Infof("ctrPrepare(%s, %s)", containerPath, ociFilename)
+// SnapshotPrepare prepares a writable snapshot from an OCI layers bundle
+// We always do it from scratch all the way, ignoring any existing state
+// that may have accumulated (like existing snapshots being avilable, etc.)
+// This effectively voids any kind of caching, but on the flip side frees us
+// from cache invalidation
+func SnapshotPrepare(containerPath string, ociFilename string) error {
+	log.Infof("snapshotPrepare(%s, %s)", containerPath, ociFilename)
 	containerID := filepath.Base(containerPath)
-	// On device restart, the existing bundle is not deleted, we need to delete the existing bundle of the container and recreate it.
-	if isBundleExists(containerID) {
-		log.Infof("ctrPrepare: a bundle with ID: %v already exists. Cleaning existing bundle and recreating it", containerID)
-		CtrRm(containerID, true)
+	// On device restart, the existing bundle is not deleted, we need to delete the
+	// existing bundle of the container and recreate it. This is safe to run even
+	// when bundle doesn't exist
+	if SnapshotRm(containerID, true) != nil {
```

Comment:
Something doesn't make sense here. We do the following 4 lines up:

```go
	containerID := filepath.Base(containerPath)
```

So the argument to `SnapshotRm` is the ID, but then [here](https://github.com/lf-edge/eve/pull/918/files#diff-2c2f111b5de32aaf6edfacf8e1090519R115-R117) we say that the argument is a full path, and even try to get the ID out of the path.

We need to be consistent and say what the argument to `SnapshotRm` is. It should be the ID.

Reply:
Good catch. I'll update the PR with more consistent naming shortly.